pkg/client/cache/delta_fifo.go

```diff
@@ -43,13 +45,13 @@ import (
 //       affects error retrying.
 //
 // Also see the comment on DeltaFIFO.
-func NewDeltaFIFO(keyFunc KeyFunc, compressor DeltaCompressor, knownObjectKeys KeyLister) *DeltaFIFO {
+func NewDeltaFIFO(keyFunc KeyFunc, compressor DeltaCompressor, knownObjects KeyLookup) *DeltaFIFO {
```

Comment:
hmm, are you suggesting i use reflect to check if the given knownObjectKeys struct satisfies KeyGetter, and only call GetByKey if it does? 

regarding staleness, I figured it's upto the clients to trust or re-get when they find a deletemarker, since not all clients really want the most upto date state?